### PR TITLE
feat!: obtain and pass DPNS contract block height

### PR DESCRIPTION
**Pull request by shumkov**
## Issue being fixed or feature implemented

To avoid setting a wrong DPNS contract ID we need to obtain and pass to Drive the block height where the contract should be created.

## What was done?

- `platform.dpns.contractId` config options is moved to `platform.dpns.contractId`
- `platform.dpns.contract.blockHeight` option is added to config
- `platform.dpns.contract.blockHeight` is set on `platform:init` command

## How Has This Been Tested?

Running `platform:init` 

## Breaking Changes

`platform.dpns.contractId` config options is moved to `platform.dpns.contractId`

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Comment by strophy** [comment]
This doesn't work for me on `evonet-6`. With `dashd:0.16` it will not sync past block 9, and with `dashd:0.15` I get errors in abci and tendermint is restarting:
```
I[2020-10-22|20:52:01.334] Executed block                               module=state height=7 validTxs=0 invalidTxs=0
I[2020-10-22|20:52:01.340] Committed state                              module=state height=7 txs=0 appHash=
E[2020-10-22|20:52:01.396] Stopping abci.socketClient for error: TypeError: e.getErrors is not a function module=abci-client connection=consensus
E[2020-10-22|20:52:01.396] Error in proxyAppConn.EndBlock               module=state err="TypeError: e.getErrors is not a function"
panic: Failed to process committed block (8:FB1FFBB1832F76612AF37D695E319D82406A01F2487B3F6CD06CBE304AF0E7AC): TypeError: e.getErrors is not a function

goroutine 289 [running]:
github.com/tendermint/tendermint/blockchain/v0.(*BlockchainReactor).poolRoutine(0xc000ff31e0)
	/home/archie/Documents/tendermint/blockchain/v0/reactor.go:344 +0x12ff
created by github.com/tendermint/tendermint/blockchain/v0.(*BlockchainReactor).OnStart
	/home/archie/Documents/tendermint/blockchain/v0/reactor.go:118 +0x84
I[2020-10-22|20:52:02.230] Found private validator                      module=main keyFile=/tendermint/data/priv_validator_key.json stateFile=/tendermint/data/priv_validator_state.json
I[2020-10-22|20:52:02.231] Found node key                               module=main path=/tendermint/data/node_key.json
I[2020-10-22|20:52:02.231] Found genesis file                           module=main path=/tendermint/config/genesis.json
ERROR: Failed to create node: error during handshake: error on replay: Could not find results for height #8
```

**Comment by shumkov** [comment]
Hehe... It’s not ready yet :)

**Review by jawid-h — DISMISSED** [review]

File: src/listr/tasks/platform/initTaskFactory.js [hunk 1]
```diff
@@ -88,15 +91,34 @@ function initTaskFactory(
...
+          // TODO You need to make broadcast return ST
+          const stateTransition = await ctx.client.platform.contracts.broadcast(
             ctx.dataContract,
             ctx.identity,
           );
+          // Get state transition commit block height from Tenderdash
+          const tendermintRpcClient = createTenderdashRpcClient();
+
+          let stateTransitionHash = crypto.createHash('sha256')
+            .update(stateTransition.toBuffer())
```
Comment (shuplenkov) [inline src/listr/tasks/platform/initTaskFactory.js:103]: Do we need `toBuffer` here?

Comment (shumkov) [reply src/listr/tasks/platform/initTaskFactory.js:103]: Yep, https://github.com/dashevo/js-dash-sdk/pull/146

**Pull request merged**